### PR TITLE
DOCS-1031: Clarify serial_path

### PR DESCRIPTION
**Pull request by andf-viam**
Clarify and standardize `serial_path` attributes description, and JSON examples that use it.

Notes: `dmesg` was the only cross-platform approach I found that worked in all cases, [approved by Matt D](https://github.com/viamrobotics/docs/pull/1596). But Rpi (or just Linux in general) is a much more likely use case, so it definitely makes sense to include direct (simpler?) instructions for that case everywhere. This PR is my attempt to balance both: platform specific initially (x2 for macOS), then platform agnostic-generally, in the hopes of casting a wide-enough net. LMK if you'd like to see `dmesg` disappear outright, however.

### CODE REVIEW

File: docs/components/motor/dmc4000.md [hunk 1]
```diff
@@ -94,7 +97,7 @@ The following attributes are available for `DMC4000` motors:
...
-| `serial_path` | string | **Required** | The full filesystem path to the serial device, starting with <file>/dev/</file>. With your serial device connected, you can run `sudo dmesg \| grep tty` to show relevant device connection log messages, and then match the returned device name, such as `ttyS0`, to its device file, such as <file>/dev/ttyS0</file>. If you omit this attribute, Viam will attempt to automatically detect the path. |
+| `serial_path` | string | Optional | The full filesystem path to the serial device, starting with <file>/dev/</file>. To find your serial device path, first connect the serial device to your smart machine, then:<ul><li>On Linux, run <code>ls /dev/serial/by-path/\*</code> to show connected serial devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/serial/by-path/usb-0:1.1:1.0"</code>.</li><li>On macOS, run <code>ls /dev/tty\* \| grep -i usb</code> to show connected USB serial devices, <code>ls /dev/tty\*</code> to browse all devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/ttyS0"</code>.</li></ul> |
```
Comment (stevebriskin) [inline docs/components/motor/dmc4000.md:100]: In this case (for dmc4000), the driver will actually try to auto-detect if a path is not provided.

File: docs/components/motor/roboclaw.md [hunk 2]
```diff
@@ -87,7 +90,7 @@ The following attributes are available for `roboclaw` motors:
...
+| `serial_path` | string | Optional | The full filesystem path to the serial device, starting with <file>/dev/</file>. To find your serial device path, first connect the serial device to your smart machine, then:<ul><li>On Linux, run <code>ls /dev/serial/by-path/\*</code> to show connected serial devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/serial/by-path/usb-0:1.1:1.0"</code>.</li><li>On macOS, run <code>ls /dev/tty\* \| grep -i usb</code> to show connected USB serial devices, <code>ls /dev/tty\*</code> to browse all devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/ttyS0"</code>.</li></ul> |
```
Comment (stevebriskin) [inline docs/components/motor/roboclaw.md:93]: serial_path is required for this driver.

File: docs/components/movement-sensor/gps/gps-nmea-rtk-serial.md [hunk 3]
```diff
@@ -139,7 +142,7 @@ The following attributes are available for a `gps-nmea-rtk-pmtk` movement sensor
...
-| `serial_path`            | string | **Required** | The full filesystem path to the serial device, starting with <file>/dev/</file>. With your serial device connected, you can run `sudo dmesg \| grep tty` to show relevant device connection log messages, and then match the returned device name, such as `ttyS0`, to its device file, such as <file>/dev/ttyS0</file>. If you omit this attribute, Viam will attempt to automatically detect the path. |
+| `serial_path`            | string | Optional     | The full filesystem path to the serial device, starting with <file>/dev/</file>. To find your serial device path, first connect the serial device to your smart machine, then:<ul><li>On Linux, run <code>ls /dev/serial/by-path/\*</code> to show connected serial devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/serial/by-path/usb-0:1.1:1.0"</code>.</li><li>On macOS, run <code>ls /dev/tty\* \| grep -i usb</code> to show connected USB serial devices, <code>ls /dev/tty\*</code> to browse all devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/ttyS0"</code>.</li></ul> |
```
Comment (stevebriskin) [inline docs/components/movement-sensor/gps/gps-nmea-rtk-serial.md:145]: It is required

File: docs/components/movement-sensor/imu/imu-wit.md [hunk 4]
```diff
@@ -101,14 +101,17 @@ Then remove and fill in the attributes as applicable to your movement sensor, ac
...
+| `serial_path`      | string | Optional  | The full filesystem path to the serial device, starting with <file>/dev/</file>. To find your serial device path, first connect the serial device to your smart machine, then:<ul><li>On Linux, run <code>ls /dev/serial/by-path/\*</code> to show connected serial devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/serial/by-path/usb-0:1.1:1.0"</code>.</li><li>On macOS, run <code>ls /dev/tty\* \| grep -i usb</code> to show connected USB serial devices, <code>ls /dev/tty\*</code> to browse all devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/ttyS0"</code>.</li></ul> |
```
Comment (stevebriskin) [inline docs/components/movement-sensor/imu/imu-wit.md:114]: required

File: docs/components/power-sensor/renogy.md [hunk 5]
```diff
@@ -92,14 +92,17 @@ Then remove and fill in the attributes as applicable to your power sensor, accor
...
+| `serial_path` | string | Optional | The full filesystem path to the serial device, starting with <file>/dev/</file>. To find your serial device path, first connect the serial device to your smart machine, then:<ul><li>On Linux, run <code>ls /dev/serial/by-path/\*</code> to show connected serial devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/serial/by-path/usb-0:1.1:1.0"</code>.</li><li>On macOS, run <code>ls /dev/tty\* \| grep -i usb</code> to show connected USB serial devices, <code>ls /dev/tty\*</code> to browse all devices, or look for your device in the output of <code>sudo dmesg \| grep tty</code>. Example: <code>"/dev/ttyS0"</code>.</li></ul> |
```
Comment (stevebriskin) [inline docs/components/power-sensor/renogy.md:105]: Defaults to "/dev/serial0" if not provided.

**Review by stevebriskin — CHANGES_REQUESTED** [review]

**Comment by andf-viam** [comment]
Thanks much! Changes made!

File: docs/components/movement-sensor/imu/imu-wit.md [hunk 6]
```diff
@@ -101,14 +101,17 @@ Then remove and fill in the attributes as applicable to your movement sensor, ac
...
+| ------------------ | ------ | ------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (npentrel) [inline docs/components/movement-sensor/imu/imu-wit.md:113]: nit
```suggestion
<!-- prettier-ignore -->
| Name               | Type   | Inclusion    | Description |
| -------------- | ------ | ----------- | ----------- |
```

**Comment by andf-viam** [comment]
Hi @stevebriskin , let me know if you'd like to see any further changes. Thanks!

**Review by stevebriskin — APPROVED** [review]

**Pull request merged**